### PR TITLE
binary output checks for REST module

### LINKED ISSUE #3992: Add methods to test binary output for REST module.
Following on from #1884 I think it would be useful to be able to test that a binary response is as expected.

I am already using this in an extension to the module. The result is compared using a hash of the response data.

We could use base64 encoding, but this would be cumbersome with large files.

### PR DESCRIPTION
**Pull request by spikyjt**
closes  #3992

### CODE REVIEW

File: src/Codeception/Module/REST.php [hunk 1]
```diff
@@ -1219,6 +1219,34 @@ public function dontSeeXmlResponseIncludes($xml)
     }
 
     /**
+     * Checks if the hash of a binary response is exactly the same as provided.
```
Comment (DavertMik) [inline src/Codeception/Module/REST.php]: a code example would be very helpful here. You can take one from your tests and add here.

Comment (spikyjt) [inline src/Codeception/Module/REST.php]: done (3 of them!)

File: src/Codeception/Module/REST.php [hunk 2]
```diff
@@ -1219,6 +1219,58 @@ public function dontSeeXmlResponseIncludes($xml)
...
+     * <?php
+     * $fileData = file_get_contents("test_file.jpg");
+     * $I->seeBinaryResponseEquals(md5($fileData));
+     * ?>
+     * ```
+     *
+     * ```php
+     * <?php
+     * $fileData = '/9j/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/yQALCAABAAEBAREA/8wABgAQEAX/2gAIAQEAAD8A0s8g/9k='; // very small jpeg
+     * $I->seeBinaryResponseEquals(hash("sha256", base64_decode($fileData)));
```
Comment (DavertMik) [inline src/Codeception/Module/REST.php]: yep, but use `sha256` but the method still uses `md5`!

File: src/Codeception/Module/REST.php [hunk 3]
```diff
@@ -1219,6 +1219,58 @@ public function dontSeeXmlResponseIncludes($xml)
...
+     */
+    public function seeBinaryResponseEquals($hash, $algo = 'md5')
+    {
+        $responseHash = hash($algo, $this->connectionModule->_getResponseContent());
+        $this->assertEquals($hash, $responseHash);
+    }
+
+    /**
+     * Checks if the hash of a binary response is not the same as provided.
+     *
```
Comment (DavertMik) [inline src/Codeception/Module/REST.php]: and one example here, please :)

**Comment by DavertMik** [comment]
Please fix the example and I'm merging.

**Comment by DavertMik** [comment]
Ok, I will update that by myself. Thanks

**Pull request merged**